hyperoptimization/explore.py

```diff
@@ -0,0 +1,197 @@
+import hpbandster.core.result as hpres
+import hpbandster.visualization as hpvis
+import json
+import matplotlib.pyplot as plt
+from collections import Counter
+import numpy as np
+import os
+
+
+class ValidFreeEnergy:
```

Comment:
For a general hyperparameter optimization framework, it may be desirable to provide an interface that allows to implement different optimization metrics (including free energies) as generically as possible.